pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -260,7 +274,9 @@ func (c *Compactor) init(objectClient client.ObjectClient, schemaConfig config.S
 			return err
 		}
 
-		c.tableMarker, err = retention.NewMarker(retentionWorkDir, c.expirationChecker, c.cfg.RetentionTableTimeout, chunkClient, r)
```

Comment:
`tableMarker` and `expirationChecker` are now created per compaction rather than up-front